chore/check-javadoc-regressions.py

```diff
@@ -0,0 +1,272 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import sys
+from collections import Counter
+from contextlib import contextmanager
+from pathlib import Path
+from subprocess import PIPE, CalledProcessError, check_output
+from tempfile import NamedTemporaryFile
+from typing import List, Optional, Pattern
+
+_COLOR_RESET = "\033[0m"
+_COLOR_SUCCESS = "\033[92;1m"
+_COLOR_HIGHLIGHT = "\033[36;1m"
+_COLOR_DIM = "\033[2m"
+_COLOR_WARNING = "\033[91;1m"
+
+
+@contextmanager
+def temporary_path(suffix: str):
+    """
+    Creates and returns a Path to a temporary file that will be deleted when
+    the context manager exits. This is needed as NamedTemporaryFile is
+    hopelessly broken on Windows due to mandatory file locks.
+    See https://bugs.python.org/issue14243.
+    """
```

Comment:
A very minor note on [docstring conventions (PEP 257)](https://www.python.org/dev/peps/pep-0257/).

Docstrings that are "sufficiently short" go one line like so:

```python
def func():
    """A short docstring."""
```

Multi-line docstrings start with a summary line, and then have a body:

```python
def func():
    """This is a longer docstring that stretches over
    multiple lines, and starts with this summary line.

    Then, additional details are written in the body
    down here, and finally the terminating quotes are
    on a line of their own.
    """
```

Concretely, this docstring could look like so:

```suggestion
    """Creates and returns a Path to a temporary file that will be deleted
    when the context manager exits.

    This is needed as NamedTemporaryFile is hopelessly broken on Windows
    due to mandatory file locks. See https://bugs.python.org/issue14243.
    """
```

It doesn't particularly matter if you edit that for this script or not and it's not something that one must strictly adhere to when writing simple scripts, just thought I'd point it out as I spotted it.